### PR TITLE
Supplier admin

### PR DESCRIPTION
**Pull request by Deepakkumar-kaaylabs**

### CODE REVIEW

File: src/styles/supplier.scss [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+        .accordion-button {
+            background-color: #004b87;
+            color: #ffff;
+            height: '42px'
+        }
+
+        .accordion-button::after {
+            background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16' fill='%23ffff'%3e%3cpath fill-rule='evenodd' d='M1.646 4.646a.5.5 0 0 1 .708 0L8 10.293l5.646-5.647a.5.5 0 0 1 .708.708l-6 6a.5.5 0 0 1-.708 0l-6-6a.5.5 0 0 1 0-.708z'/%3e%3c/svg%3e");
+        }
```
Comment (MJyotheesh) [inline src/styles/supplier.scss:20]: images or svgs don't import in scss files

File: src/styles/supplier-details.scss [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+    .accordion-button {
+      background-color: #004b87;
+      color: #ffff;
+      height: "42px";
+    }
+
+    .accordion-button::after {
+      background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16' fill='%23ffff'%3e%3cpath fill-rule='evenodd' d='M1.646 4.646a.5.5 0 0 1 .708 0L8 10.293l5.646-5.647a.5.5 0 0 1 .708.708l-6 6a.5.5 0 0 1-.708 0l-6-6a.5.5 0 0 1 0-.708z'/%3e%3c/svg%3e");
+    }
```
Comment (MJyotheesh) [inline src/styles/supplier-details.scss:23]: images or svgs don't import in scss files

File: src/styles/loginPage.scss
Comment (MJyotheesh) [inline src/styles/loginPage.scss:1]: Change file name as login-page.scss

File: src/shared/meta-data/lookups.ts [hunk 3]
```diff
@@ -10,7 +10,8 @@ export default {
           required: "Please Enter Lookup Value",
         },
       ],
-      submitButtonText:"Add"
+      submitButtonText:"Add",
+      notificationMessage:"Lookup Value Added Successfully"
```
Comment (MJyotheesh) [inline src/shared/meta-data/lookups.ts:14]: get from const validationMessages

File: src/shared/meta-data/lookups.ts [hunk 4]
```diff
@@ -22,12 +23,14 @@ export default {
           required: "Please Enter Lookup Value",
         },
       ],
-        submitButtonText: "Update"
+        submitButtonText: "Update",
+      notificationMessage: "Lookup Value Updated Successfully"
```
Comment (MJyotheesh) [inline src/shared/meta-data/lookups.ts:27]: get from const validationMessages

File: src/shared/meta-data/lookups.ts [hunk 5]
```diff
@@ -22,12 +23,14 @@ export default {
...
     },
     remove: {
       title: "Delete Lookup Value",
       body: "Are you sure you want to remove this item?",
-      submitButtonText: "Delete"
+      submitButtonText: "Delete",
+      notificationMessage: "Lookup value Deleted Successfully"
```
Comment (MJyotheesh) [inline src/shared/meta-data/lookups.ts:33]: get from const validationMessages

File: src/shared/meta-data/lookups.ts [hunk 6]
```diff
@@ -39,7 +42,8 @@ export default {
           required: "input",
         },
       ],
-        submitButtonText: "Create"
+        submitButtonText: "Create",
+      notificationMessage: "Lookup Created Succussfully"
```
Comment (MJyotheesh) [inline src/shared/meta-data/lookups.ts:46]: get from const validationMessages

File: src/shared/meta-data/index.ts [hunk 7]
```diff
@@ -1,3 +1,3 @@
 import lookups from "./lookups";
-
-export { lookups };
+import supplierList from "./supplier-list";
+export { lookups, supplierList };
```
Comment (MJyotheesh) [inline src/shared/meta-data/index.ts:3]: give one line space after the imports in all the places

File: src/shared/icon-helper/index.tsx [hunk 8]
```diff
@@ -21,6 +21,8 @@ import { RightArrow } from './right-arrow';
 import { SearchIcon } from './search-icon';
 import { DropDownArrow } from './drop-down-arrow';
 import { LeftArrow } from './left-arrow';
+import ImageFileIcon from './img-icon';
+import { NoDataIcon } from './no-data-icon';
```
Comment (MJyotheesh) [inline src/shared/icon-helper/index.tsx:28]: why unused lines?

File: src/shared/constant-values.ts [hunk 9]
```diff
@@ -28,3 +28,15 @@ export const lookupApiParams = {
...
+
+export const mobileScreenWidth = 500;
+
+export enum Status {
+  APPROVED = "Approved",
+  UNAPPROVED = "Unapproved",
+  REJECTED = "Rejected",
+}
+
+export const statusList = ["Approved", "Rejected", "Unapproved"];
```
Comment (MJyotheesh) [inline src/shared/constant-values.ts:40]: use any one method right

File: src/shared/constant-values.ts [hunk 10]
```diff
@@ -28,3 +28,15 @@ export const lookupApiParams = {
 export const defaultData = [
   { lookupName: "", lookupValues: [], lookupUid: "" },
 ];
+
+export const mobileScreenWidth = 500;
```
Comment (MJyotheesh) [inline src/shared/constant-values.ts:32]: 570px

File: src/pages/supplier/supplier-profile.tsx [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+const SupplierProfile = (props: SupplierProfileProps) => {
+  const { supplierUid } = props;
+  const [supplierData, setSupplierData] = useState<any>({});
+  const [loading, setLoading] = useState(false);
+  const fetchSupplierData = async () => {
+    try {
+      setLoading(true);
+      const data = await getSupplier({ supplierUid, userType: role });
+      setSupplierData(data.result[0]);
```
Comment (MJyotheesh) [inline src/pages/supplier/supplier-profile.tsx:24]: get

File: src/pages/supplier/supplier-profile.tsx [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+import { supplierList } from "../../shared/meta-data";
+const {
+  supplierProfile: { fileList },
+} = supplierList;
+import { role } from "../../shared/constant-values";
+import { Spinner } from "../../components/generics";
+
+type SupplierProfileProps = {
+  supplierUid: string;
+};
```
Comment (MJyotheesh) [inline src/pages/supplier/supplier-profile.tsx:14]: move all types to types.ts file (Refer supplier web)

File: src/pages/supplier/supplier-profile.tsx [hunk 13]
```diff
@@ -0,0 +1,120 @@
+import React, { useEffect, useState } from "react";
+import { Accordion, Col, Row, Card } from "react-bootstrap";
+import { ImageFileIcon } from "shared/icon-helper";
+import { getSupplier } from "../../shared/api/url-helper";
+import { supplierList } from "../../shared/meta-data";
+const {
+  supplierProfile: { fileList },
+} = supplierList;
```
Comment (MJyotheesh) [inline src/pages/supplier/supplier-profile.tsx:8]: declare after imports

File: src/pages/supplier/supplier-details.tsx [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+  const { supplierUid: paramsUid = "" } = useParams();
+  const [supplierUid, setSupplierUid] = useState(propsUid || paramsUid);
+
+  useEffect(() => {
+    setSupplierUid(propsUid || paramsUid);
+    ref.current.scroll({ top: "10px", behavior: "smooth" });
+  }, [propsUid]);
+  return (supplierUid &&
+    <div className="supplier-details-container px-md-0 px-sm-0 px-1" ref={ref}>
+      {mobileScreenWidth>=window.innerWidth && <NavLink to='/supplier'>Back</NavLink>}
```
Comment (MJyotheesh) [inline src/pages/supplier/supplier-details.tsx:27]: window.innerwidth declare in App.tsx and use it like this

useEffect(() => {
    const windowWidth = get(window, "innerWidth", 0);
    if (windowWidth < mobileScreenWidth) {
      dispatch(actions.setMobileView(true));
      setMobileView(true);
    } else {
      dispatch(actions.setMobileView(false));
      setMobileView(false);
    }
  }, []);

File: src/pages/supplier/index.tsx [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+      setloading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchSupplierData();
+  }, []);
+
+  useEffect(() => {
+    setisMobile(mobileScreenWidth >= window.innerWidth);
```
Comment (MJyotheesh) [inline src/pages/supplier/index.tsx:38]: move to app.tsx

File: src/pages/supplier/index.tsx [hunk 16]
```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from "react";
+import SupplierList from "./supplier-list";
+import SupplierDetails from "./supplier-details";
+import { getSupplier } from "../../shared/api/url-helper";
+import { NoDataIcon } from "shared/icon-helper";
+import { useNavigate } from "react-router-dom";
+import { mobileScreenWidth } from "../../shared/constant-values";
+import { Spinner, TitleWithSearch } from "../../components/generics";
```
Comment (MJyotheesh) [inline src/pages/supplier/index.tsx:8]: TitleSearch

File: src/pages/lookups/index.tsx [hunk 17]
```diff
@@ -97,12 +102,10 @@ const Lookups = () => {
   return (
     <div className="container-fluid lookups p-10">
       <div className="lookup-header">
-        <div className="row mt-2">
-          <div className="col-xxl-6 col-xl-6 col-sm-6 col-6 title">Lookups</div>
-          <div className="col-xxl-6 col-xl-6 col-sm-6 col-6 d-flex justify-content-end">
-            <input type="search" />
-          </div>
-        </div>
+        <TitleWithSearch
```
Comment (MJyotheesh) [inline src/pages/lookups/index.tsx:105]: TitleSearch

File: src/pages/driver/index.tsx [hunk 18]
```diff
@@ -1,9 +1,78 @@
...
 const Driver = () => {
+  const [driverData, setDriverData] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [activeTab, setActiveTab] = useState(statusList[0]);
+
+  const fetchDriverData = async () => {
+    try {
+      setLoading(true);
+      const res = await getDriversList({ status: activeTab, userType: role });
+      setDriverData(res.result);
```
Comment (MJyotheesh) [inline src/pages/driver/index.tsx:29]: get

**Review by MJyotheesh — CHANGES_REQUESTED** [review]
Resolve Comments

File: src/pages/driver/index.tsx (hunk 18 above)
Comment (Deepakkumar-kaaylabs) [reply src/pages/driver/index.tsx:29]: we are not using lodash in admin-web

Comment (MJyotheesh) [reply src/pages/driver/index.tsx:29]: Not lodash refer supplier web for get function

File: src/styles/supplier.scss (hunk 1 above)
Comment (Deepakkumar-kaaylabs) [reply src/styles/supplier.scss:20]: In accordion, this is the only way to change the collapse button , as we discussed will create custom accordion component then will remove this import bro

File: src/pages/supplier/supplier-details.tsx (hunk 14 above)
Comment (Deepakkumar-kaaylabs) [reply src/pages/supplier/supplier-details.tsx:27]: We are not implemented redis in admin once we done that will change it

Comment (MJyotheesh) [reply src/pages/supplier/supplier-details.tsx:27]: U can copy from supplier web right

**Review by MJyotheesh — APPROVED** [review]

**Pull request merged**